mapper/orm/src/main/java/org/hibernate/search/mapper/orm/loading/impl/HibernateOrmMassEntityLoader.java

```diff
@@ -62,4 +57,24 @@ public void load(List<I> identifiers) throws InterruptedException {
 		}
 		transactionHelper.commit( session );
 	}
+
+	@SuppressWarnings("unchecked") // We can assume identifiers are serializable
+	private List<E> multiLoad(List<I> identifiers) {
+		return typeQueryLoader.createMultiIdentifierLoadAccess( session )
+				.with( options.cacheMode() )
+				.with( LockOptions.NONE )
+				.multiLoad( (List<Serializable>) identifiers );
+	}
+
+	private List<E> queryByIds(List<I> identifiers) {
+		return typeQueryLoader.createLoadingQuery( session, ID_PARAMETER_NAME )
+				.setParameter( ID_PARAMETER_NAME, identifiers )
+				.setCacheMode( options.cacheMode() )
+				.setLockOptions( LockOptions.NONE )
+				.setCacheable( false )
+				.setHibernateFlushMode( FlushMode.MANUAL )
+				.setFetchSize( identifiers.size() )
+				.setHint( QueryHints.HINT_CACHEABLE, true )
```

Comment:
I don't think this is necessary? And it's probably harmful for performance, since we will cache query results but never use the query cache.

Also, it directly contractics the `setCacheable( false )` just above.

```suggestion
```

Reply:
I'm doing the change, but probably this flag would help in hitting the second level cache in the case I mentioned above:
1. property id != document id
2. entity has some lazy fields
But without a test we cannot say that for sure!
Moreover, we could handle caching in case the document id was the natural id, using the natural id multiloading.